File: 0242-valid-anagram/0242-valid-anagram.go
func isAnagram(s string, t string) bool {
    if (len(s) != len(t)) {
        return false
    }

    arr := [26]int{}

    for i := 0; i < len(s); i++ {
        arr[s[i] - 'a']++
        arr[t[i] - 'a']--
    }

    for _, val := range arr {
        if val != 0 {
            return false
        }
    }

    return true
}

File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.size() != t.size()) return false;

        vector<int> arr(26, 0);

        for (int i = 0; i < s.size(); i++) {
            arr[s[i]  - 'a']++;
            arr[t[i] - 'a']--;
        }

        for (int num : arr) {
            if (num != 0) return false;
        }

        return true;
    }
};